docs/subnets/introduction-to-vm.md

```diff
@@ -0,0 +1,335 @@
+# Introduction to Virtual Machines
+
+## Overview
+
+This is part of a series of tutorials for building a Virtual Machine (VM):
+
+- Introduction to Virtual Machines (this article)
+- [How to Build a Simple VM](./create-a-virtual-machine-vm.md)
+- [How to Build a Complex VM](./create-a-vm-blobvm.md)
+
+A Virtual Machine is a blueprint for a blockchain and a blockchain instantiates from a VM, very much similar to a class-object relationship. A Virtual Machine can define following: issuance of transactions, transaction types, block structure, block building algorithm, keeping pending transactions in the mempool, gossiping transactions to the connected nodes, etc.
+
+## Blocks and State
+
+Virtual Machines can be broken down into 2 components: blocks and state. The functionality provided by VMs is to:
+
+- Define the representation of a blockchain's state
+- Represent the operations in that state
+- Apply the operations in that state
+
+Each block in the blockchain contains a set of state transitions. Each block is applied in order from the blockchain's initial genesis block to its last accepted block to reach the latest state of the blockchain.
+## Blockchain
+
+A blockchain has 2 components: **Consensus Engine** and **Virtual Machine**. VMs mainly deal with the implementation related to the block's structure: building and parsing. The Consensus Engine helps in reaching consensus on the block issued by VM. Here is a brief overview:
+
+1. A validator node wants to update the blockchain's state
+2. It creates a block with state transition details in it
```

Comment:
Can we improve the phrase state transition details?

Reply:
We should be more specific with the subject of the sentence and never use something like `It`. We should be painfully specific here to make sure it's clear.